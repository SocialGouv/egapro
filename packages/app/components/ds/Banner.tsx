import * as React from "react"
import {
  CloseButton,
  HStack,
  Icon,
  Square,
  Text,
  useBreakpointValue,
  useColorModeValue,
  useDisclosure,
} from "@chakra-ui/react"
import { FiInfo } from "react-icons/fi"

export const Banner = () => {
  const { isOpen, onClose, onOpen } = useDisclosure()
  const isMobile = useBreakpointValue({ base: true, md: false })
  const bgColor = useColorModeValue("yellow.100", "yellow.400")

  React.useEffect(() => {
    // First case : on mount, the variable in sessionStorage is undefined.
    if (sessionStorage.getItem("banner.dontShow") === null || sessionStorage.getItem("banner.dontShow") === "false")
      onOpen()
    // Second case : the variable in sessionStorage exists and the user opted to don't show the banner.
    if (sessionStorage.getItem("banner.dontShow") === "true") onClose()
  }, [onClose, onOpen])

  if (!isOpen) return null

  function handleClose() {
    sessionStorage.setItem("banner.dontShow", "true")
    onClose()
  }

  return (
    <HStack
      justify="space-between"
      border="1px solid"
      borderColor="yellow.300"
      px="2"
      mt="4"
      borderRadius="lg"
      bgColor={bgColor}
      textColor={"gray.800"}
    >
      <HStack spacing="2" align="center">
        {!isMobile && (
          <Square size="12" borderRadius="md">
            <Icon as={FiInfo} boxSize="6" />
          </Square>
        )}
        <Text fontWeight="medium" textColor="gray.800">
          Les informations de l'entreprise ou de l'UES ainsi que la tranche d'effectifs, correspondent à la dernière
          déclaration.
        </Text>
      </HStack>
      <CloseButton aria-label="Fermer le panneau d'information" onClick={handleClose} />
    </HStack>
  )
}
