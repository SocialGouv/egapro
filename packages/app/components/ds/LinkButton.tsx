import React from "react"
import { Flex, LinkBox, LinkOverlay, Spacer } from "@chakra-ui/react"

type LinkButtonProps = {
  children: React.ReactNode
  color?: string
  href: string
  isExternal?: boolean
  leftIcon?: React.ReactNode
}

export function LinkButton({
  children,
  color = "primary.500",
  href,
  isExternal = false,
  leftIcon = null,
  ...rest
}: LinkButtonProps) {
  return (
    <LinkBox>
      <LinkOverlay href={href} isExternal={isExternal}>
        <Flex
          justify="center"
          align="center"
          border="1px solid"
          width="fit-content"
          px={3}
          py={2}
          borderRadius="lg"
          borderColor={color}
          margin="auto"
          {...rest}
        >
          {leftIcon}
          <Spacer ml={2} />
          {children}
        </Flex>
      </LinkOverlay>
    </LinkBox>
  )
}
