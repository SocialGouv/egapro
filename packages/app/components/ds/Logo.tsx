import React, { FunctionComponent } from "react"
import { Image } from "@chakra-ui/react"

export type LogoProps = Record<string, unknown>

const Logo: FunctionComponent<LogoProps> = () => {
  return (
    <Image
      src={"/consulter-index/icons/marianne.svg"}
      alt="Aller à la page d'accueil du Ministère du travail, de l'emploi et de l'insertion"
      width={88}
    />
  )
}

export default Logo
