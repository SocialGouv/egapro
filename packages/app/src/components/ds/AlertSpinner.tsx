import { Alert, AlertTitle, Spinner } from "@chakra-ui/react";
import type { PropsWithChildren } from "react";

export const AlertSpinner = ({ children }: PropsWithChildren) => {
  return (
    <Alert
      status="info"
      variant="subtle"
      flexDirection="column"
      alignItems="center"
      justifyContent="center"
      textAlign="center"
      height="200px"
      mt={4}
      colorScheme="gray"
    >
      <Spinner thickness="4px" speed="0.65s" emptyColor="gray.200" color="blue.500" size="lg" />
      <AlertTitle mt={4} mb={1} fontSize="lg">
        {children}
      </AlertTitle>
    </Alert>
  );
};
