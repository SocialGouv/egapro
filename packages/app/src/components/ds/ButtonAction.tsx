import type { ButtonProps as ChakraButtonProps } from "@chakra-ui/react";
import { Button } from "@chakra-ui/react";

export type ButtonProps = ChakraButtonProps & {
  colorScheme?: "gray" | "orange" | "primary";
  fullWidth?: boolean;
  label: string;
  leftIcon?: React.ReactElement;
  rightIcon?: React.ReactElement;
  size?: "lg" | "md" | "sm" | "xs";
  variant?: "ghost" | "link" | "outline" | "solid";
};

export type ButtonActionProps = ButtonProps & {
  disabled?: boolean;
  loading?: boolean;
  onClick?: () => void;
  type?: "button" | "reset" | "submit" | undefined;
};

export const ButtonAction = ({
  label,
  colorScheme = "primary",
  variant = "solid",
  onClick,
  leftIcon,
  rightIcon,
  type,
  size = "md",
  disabled = false,
  loading = false,
  fullWidth,
  ...rest
}: ButtonActionProps) => (
  <Button
    size={size}
    type={type}
    onClick={onClick}
    isDisabled={disabled}
    isLoading={loading}
    colorScheme={colorScheme}
    variant={variant}
    leftIcon={leftIcon}
    rightIcon={rightIcon}
    sx={{
      width: fullWidth ? "100%" : "auto",
    }}
    {...rest}
  >
    {label}
  </Button>
);
