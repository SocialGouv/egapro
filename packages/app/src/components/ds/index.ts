export * from "./ActivityIndicator";
export * from "./AlertSpinner";
export * from "./Banner";
export * from "./ButtonAction";
export * from "./Footer";
export * from "./Header";
export * from "./InputGroup";
export * from "./LinkButton";
export * from "./Logo";
export * from "./SinglePageLayout";
export * from "./TextLink";
