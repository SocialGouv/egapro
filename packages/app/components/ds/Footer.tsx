import React from "react"
// import NextLink from "next/link"
import { Box, Container, List, ListItem, Text, Link, Flex } from "@chakra-ui/react"

import Logo from "@/components/ds/Logo"
import TextLink from "@/components/ds/TextLink"
import packageConfig from "../../package.json"
import { ExternalLinkIcon } from "@chakra-ui/icons"

function Footer() {
  const version = process.env.REACT_APP_VERSION || packageConfig.version

  return (
    <Box
      id="footer"
      as="footer"
      role="contentinfo"
      py={6}
      sx={{
        borderTop: "1px solid #E3E4ED",
        "@media print": {
          display: "none",
        },
      }}
    >
      <Container
        maxW="container.xl"
        sx={{
          "@media (min-width: 1024px)": {
            display: "flex",
            flexDirection: "row",
          },
        }}
      >
        <Box
          sx={{
            "@media (min-width: 640px)": {
              display: "flex",
              flexDirection: "row",
            },
          }}
        >
          <Box pr={8}>
            <Link href="https://travail-emploi.gouv.fr/" isExternal pt={2} sx={{ display: "block" }}>
              <Logo />
            </Link>
          </Box>

          <List
            fontSize="sm"
            sx={{
              "@media (max-width: 639px)": {
                marginTop: "var(--chakra-space-4)",
              },
            }}
          >
            <ListItem>
              <Link href="https://travail-emploi.gouv.fr/IMG/xlsx/referents_egalite_professionnelle.xlsx" isExternal>
                <Flex justify="center" align="center">
                  Télécharger la liste des référents Egapro (XLSX, 22 Ko) <ExternalLinkIcon ml={2} />
                </Flex>
              </Link>
            </ListItem>
            {/* <ListItem>
              <NextLink href="/mentions-legales">
                <Link>Mentions légales</Link>
              </NextLink>
            </ListItem>
            <ListItem>
              <Link href={process.env.PUBLIC_URL + "/a11y/declaration-accessibilite-rgaa-4-1.pdf"}>
                Accessibilité&nbsp;: partiellement conforme
              </Link>
            </ListItem>
            <ListItem>
              <NextLink href="/cgu">
                <Link>Conditions générales d'utilisation</Link>
              </NextLink>
            </ListItem>
            <ListItem>
              <NextLink href="/politique-confidentialite">
                <Link>Politique de confidentialité</Link>
              </NextLink>
            </ListItem> */}
          </List>
        </Box>

        <Box
          sx={{
            "@media (max-width: 1023px)": {
              marginTop: "var(--chakra-space-4)",
            },
            "@media (min-width: 1024px)": {
              marginLeft: "auto",
            },
          }}
        >
          <Text fontSize="sm" as="i">
            Index Egapro a été développé par les équipes de la fabrique numérique des ministères sociaux
          </Text>
          <Text fontSize="sm" mt={1}>
            Pour nous aider à l'améliorer{" "}
            <TextLink
              to="https://voxusagers.numerique.gouv.fr/Demarches/2240?&view-mode=formulaire-avis&nd_mode=en-ligne-enti%C3%A8rement&nd_source=button&key=73366ddb13d498f4c77d01c2983bab48"
              isExternal
            >
              donnez-nous votre avis
            </TextLink>
            {", "}
            <TextLink to={`https://github.com/SocialGouv/egapro/tree/${version}`} isExternal>
              contribuez sur Github
            </TextLink>
          </Text>
        </Box>
      </Container>
    </Box>
  )
}

export default Footer
