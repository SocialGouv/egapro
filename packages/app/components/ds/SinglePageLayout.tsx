import React, { ReactNode } from "react"
import { Box, Container, Flex } from "@chakra-ui/layout"

import Header from "@/components/ds/Header"
import Footer from "@/components/ds/Footer"

interface Props {
  children: ReactNode
}

export function SinglePageLayout({ children }: Props) {
  return (
    <Flex direction="column" minHeight="100vh">
      <Header />
      <Box as="main" role="main" id="main" flexGrow={1} pt={10}>
        <Container maxW="container.lg">{children}</Container>
      </Box>
      <Footer />
    </Flex>
  )
}
