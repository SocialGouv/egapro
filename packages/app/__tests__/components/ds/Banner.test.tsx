import { render } from "@testing-library/react"
import { Banner } from "@/components/ds/Banner"

beforeAll(() => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
})

test("should match snapshot", () => {
  const view = render(<Banner />)
  expect(view).toMatchSnapshot()
})
