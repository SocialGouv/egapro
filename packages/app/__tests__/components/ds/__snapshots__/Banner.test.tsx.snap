// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <div
        class="chakra-stack css-ndtc64"
      >
        <div
          class="chakra-stack css-g9cw6v"
        >
          <p
            class="chakra-text css-7wnz5b"
          >
            Les informations de l'entreprise ou de l'UES ainsi que la tranche d'effectifs, correspondent à la dernière déclaration.
          </p>
        </div>
        <button
          aria-label="Fermer le panneau d'information"
          class="css-4abrhy"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-stack css-ndtc64"
    >
      <div
        class="chakra-stack css-g9cw6v"
      >
        <p
          class="chakra-text css-7wnz5b"
        >
          Les informations de l'entreprise ou de l'UES ainsi que la tranche d'effectifs, correspondent à la dernière déclaration.
        </p>
      </div>
      <button
        aria-label="Fermer le panneau d'information"
        class="css-4abrhy"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
